testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketShutdownOutputBySelfTest.java

```diff
@@ -140,5 +219,22 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
         public void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
             queue.offer(msg.readByte());
         }
+
+        private void drainWritabilityQueue() throws InterruptedException {
+            while ((writabilityQueue.poll(100, TimeUnit.MILLISECONDS)) != null) {
+                // Just drain the queue.
+            }
+        }
+
+        void assertWritability(boolean isWritable) throws InterruptedException {
+            try {
+                Boolean writability = writabilityQueue.takeLast();
+                assertEquals(isWritable, writability);
+                // TODO(scott): why do we get multiple writability changes here ... race condition?
+                drainWritabilityQueue();
+            } catch (Throwable c) {
+                c.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1148)